Automattic/jetpack
# Protect: Add ability to manually enqueue scans

#### Description
This PR adds the ability for paid users to manually enqueue scans in Jetpack Protect.

#### Changes proposed in this Pull Request:

* Adds a `/scan` endpoint to the Jetpack REST API
* Adds a `scan` method to the `Threats` class for enqueuing scans
* Adds the ability to enqueue a scan by sending a request to the wpcom API, via the `scan()` action.
* Integrates all of the above into the paid threats list

#### Other information:

#### Jetpack product discussion


- 1201069996155224-as-1203034583317307

#### Does this pull request change what data or activity we track or use?


- No

#### Testing instructions:





* Start up a fresh JN site with Jetpack Beta, running this branch of Protect
* Install a threat - a vulnerable plugin such as [WooCommerce 3.0.0](https://downloads.wordpress.org/plugin/woocommerce.3.0.0.zip)
* Get started with Jetpack Protect free
* Validate no regression in the admin dashboard (free users should not be able to enqueue scans)
* Upgrade to a Jetpack plan that includes Scan
* Validate that each manual scan button enqueues a new scan
* Verify from Calypso and/or RWDB that the scan has begun, and progresses/succeeds as expected
* Ensure that the Protect UI handles as expected throughout the scanning process and that the status report displayed includes the results of the latest Scan

#### Screenshots:
<img width="1185" alt="Screen Shot 2022-10-12 at 13 27 25" src="https://user-images.githubusercontent.com/43220201/195441238-e9d0c5db-65f8-4682-8b8d-801597f9f300.png">
